src/service/cid-impl.js

```diff
@@ -182,15 +182,15 @@ function getOrCreateCookie(cid, getCidStruct, persistenceConsent) {
   const newCookie = 'amp-' + cid.sha384Base64_(getEntropy(win));
 
   // Store it as a cookie based on the persistence consent.
-  persistenceConsent.then(() => {
+  return persistenceConsent.then(() => {
```

Comment:
Can we add a line before
`this.baseCid_ = newCookie`

and the same for `localStorage`.